spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java

```diff
@@ -253,26 +255,4 @@ public void afterPropertiesSet() {
 				+ " services) , discovery client is now available");
 	}
 
-	private boolean matchServiceLabels(V1Service service) {
```

Comment:
let's remove this method and move it to a new Utils: `KubernetesDiscoveryClientUtils`. Also I've simplified the code slightly and added proper tests